File: source_funcs/go_transcoder/MINIMUM_COST_CONNECT_WEIGHTED_NODES_REPRESENTED_ARRAY/MINIMUM_COST_CONNECT_WEIGHTED_NODES_REPRESENTED_ARRAY.go
package main

import "math"

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}

func f_gold(a []int, n int) int {
	var (
		mn  int = math.MaxInt64
		sum int = 0
	)
	for i := int(0); i < n; i++ {
		mn = min(a[i], mn)
		sum += a[i]
	}
	return mn * (sum - mn)
}

File: rust_funcs/cpp_transcoder/MINIMUM_COST_CONNECT_WEIGHTED_NODES_REPRESENTED_ARRAY/MINIMUM_COST_CONNECT_WEIGHTED_NODES_REPRESENTED_ARRAY.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


fn f_gold(a: [f32;2], n: i32) -> i32 {
    let mut mn = f32::MAX; 
    let mut sum = 0.0;
    for i in 0..n {
        mn = a[i as usize].min(mn);
        sum += a[i as usize];
    }
    mn * (sum - mn)
}
